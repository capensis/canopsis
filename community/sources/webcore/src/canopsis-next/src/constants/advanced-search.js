export const ADVANCED_SEARCH_ITEM_TYPES = {
  field: 'field',
  condition: 'condition',
  value: 'value',
  union: 'union',
};

export const ADVANCED_SEARCH_NEXT_ITEM_TYPES = {
  [ADVANCED_SEARCH_ITEM_TYPES.field]: ADVANCED_SEARCH_ITEM_TYPES.condition,
  [ADVANCED_SEARCH_ITEM_TYPES.condition]: ADVANCED_SEARCH_ITEM_TYPES.value,
  [ADVANCED_SEARCH_ITEM_TYPES.value]: ADVANCED_SEARCH_ITEM_TYPES.union,
  [ADVANCED_SEARCH_ITEM_TYPES.union]: ADVANCED_SEARCH_ITEM_TYPES.field,
};

export const ADVANCED_SEARCH_NOT = 'NOT';

export const ADVANCED_SEARCH_UNION_CONDITIONS = {
  and: 'AND',
  or: 'OR',
};

export const ADVANCED_SEARCH_CONDITIONS = {
  less: '<',
  more: '>',
  equal: '=',
  notEqual: '!=',
  like: 'LIKE',
  contains: 'CONTAINS',
};

export const ADVANCED_SEARCH_UNION_REGEXP_PATTERN = new RegExp(`\\s(${Object.values(ADVANCED_SEARCH_UNION_CONDITIONS).join('|')})(\\s|$)`, 'gi');

export const ADVANCED_SEARCH_UNION_FIELDS = [
  {
    value: ADVANCED_SEARCH_UNION_CONDITIONS.and,
    type: ADVANCED_SEARCH_ITEM_TYPES.union,
    text: ADVANCED_SEARCH_UNION_CONDITIONS.and,
  },
  {
    value: ADVANCED_SEARCH_UNION_CONDITIONS.or,
    type: ADVANCED_SEARCH_ITEM_TYPES.union,
    text: ADVANCED_SEARCH_UNION_CONDITIONS.or,
  },
];
