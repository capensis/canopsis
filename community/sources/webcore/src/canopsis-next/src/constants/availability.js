import { QUICK_RANGES } from './common';

export const AVAILABILITY_SHOW_TYPE = {
  percent: 0,
  duration: 1,
};

export const AVAILABILITY_QUICK_RANGES = {
  [QUICK_RANGES.last1Hour.value]: QUICK_RANGES.last1Hour,
  [QUICK_RANGES.last24Hour.value]: QUICK_RANGES.last24Hour,
  [QUICK_RANGES.today.value]: QUICK_RANGES.today,
  [QUICK_RANGES.yesterday.value]: QUICK_RANGES.yesterday,
  [QUICK_RANGES.last7Days.value]: QUICK_RANGES.last7Days,
  [QUICK_RANGES.previousWeek.value]: QUICK_RANGES.previousWeek,
  [QUICK_RANGES.previousMonth.value]: QUICK_RANGES.previousMonth,
  [QUICK_RANGES.thisMonth.value]: QUICK_RANGES.thisMonth,
  [QUICK_RANGES.thisMonthSoFar.value]: QUICK_RANGES.thisMonthSoFar,
  [QUICK_RANGES.last3Months.value]: QUICK_RANGES.last3Months,
  [QUICK_RANGES.last6Months.value]: QUICK_RANGES.last6Months,
};
