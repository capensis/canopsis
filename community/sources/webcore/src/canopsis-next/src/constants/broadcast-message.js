export const DEFAULT_BROADCAST_MESSAGE_COLOR = '#e75e40';

export const BROADCAST_MESSAGES_STATUSES = {
  active: 0,
  pending: 1,
  expired: 2,
};
