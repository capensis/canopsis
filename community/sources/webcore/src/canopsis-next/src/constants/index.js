export * from './recurrence-rule';
export * from './common';
export * from './entity';
export * from './alarm';
export * from './broadcast-message';
export * from './date';
export * from './dynamic-info';
export * from './event-filter';
export * from './pattern';
export * from './healthcheck';
export * from './idle-rule';
export * from './junit';
export * from './modal';
export * from './pbehavior';
export * from './permission';
export * from './remediation';
export * from './scenario';
export * from './snmp';
export * from './stats';
export * from './store';
export * from './widget';
export * from './kpi';
export * from './user';
export * from './service-weather';
export * from './map';
export * from './flowchart';
export * from './parameters';
export * from './declare-ticket';
export * from './link-rule';
export * from './external-data';
export * from './tag';
export * from './counter';
export * from './calendar';
export * from './icon';
export * from './availability';
