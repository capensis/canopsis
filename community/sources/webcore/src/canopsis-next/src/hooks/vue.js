import { getCurrentInstance } from 'vue';

/**
 * Hook for get current component instance
 *
 * @return {Vue}
 */
export const useComponentInstance = () => {
  const instance = getCurrentInstance();

  return instance?.proxy || instance;
};

/**
 * Hook for get current component options
 *
 * @return {Object}
 */
export const useComponentOptions = () => {
  const instance = useComponentInstance();

  return instance.$options;
};

/**
 * Hook for get current component model
 *
 * @return {Object}
 */
export const useComponentModel = () => {
  const { model } = useComponentOptions();

  return {
    event: model?.event ?? 'input',
    prop: model?.event ?? 'value',
  };
};
