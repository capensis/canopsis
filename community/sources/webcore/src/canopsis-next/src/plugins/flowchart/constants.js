export const DIRECTIONS = {
  northWest: 'nw',
  north: 'n',
  northEast: 'ne',
  east: 'e',
  southEast: 'se',
  south: 's',
  southWest: 'sw',
  west: 'w',
};
