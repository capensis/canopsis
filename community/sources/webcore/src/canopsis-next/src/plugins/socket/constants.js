export const REQUEST_MESSAGES_TYPES = {
  ping: 0,
  join: 1,
  leave: 2,
  authenticate: 3,
};

export const RESPONSE_MESSAGES_TYPES = {
  pong: 0,
  ok: 1,
  error: 2,
  close: 3,
  authenticated: 4,
};

export const ICONS_RESPONSE_MESSAGES_TYPES = {
  create: 0,
  update: 1,
  delete: 2,
};

export const MAX_RECONNECTS_COUNT = 10;

export const PING_INTERVAL = 5000;

export const RECONNECT_INTERVAL = 5000;

export const EVENTS_TYPES = {
  customClose: 'custom-close',
  closeRoom: 'close-room',
  networkError: 'network-error',
};

export const ERROR_MESSAGES = {
  authenticationFailed: 'authentication failed',
};
