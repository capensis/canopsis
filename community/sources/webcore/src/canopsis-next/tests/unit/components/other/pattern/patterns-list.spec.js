import { range } from 'lodash';

import { flushPromises, generateRenderer } from '@unit/utils/vue';
import {
  selectRowRemoveButtonByIndex,
  selectRowEditButtonByIndex,
  selectRowCheckboxByIndex,
  selectMassRemoveButton,
} from '@unit/utils/table';

import { PATTERN_TYPES } from '@/constants';

import PatternsList from '@/components/other/pattern/patterns-list.vue';
import CAdvancedDataTable from '@/components/common/table/c-advanced-data-table.vue';

const stubs = {
  'c-advanced-data-table': CAdvancedDataTable,
  'c-search-field': true,
  'v-checkbox': true,
  'c-expand-btn': true,
  'c-action-btn': true,
  'c-table-pagination': true,
};

describe('patterns-list', () => {
  const totalItems = 11;

  const types = [PATTERN_TYPES.alarm, PATTERN_TYPES.entity, PATTERN_TYPES.pbehavior];

  const patterns = range(totalItems).map(index => ({
    _id: `pattern-id-${index}`,
    type: types[index % 3],
    enabled: !!(index % 2),
    author: {
      display_name: `author-${index}`,
    },
    updated: 1614861888 + index,
  }));

  const snapshotFactory = generateRenderer(PatternsList, { stubs });

  it('Selected items removed after trigger mass remove button', async () => {
    const wrapper = snapshotFactory({
      propsData: {
        patterns,
        options: {
          page: 1,
          itemsPerPage: 10,
        },
        totalItems,
      },
    });

    await flushPromises();

    await selectRowCheckboxByIndex(wrapper, 0).trigger('click');
    await selectRowCheckboxByIndex(wrapper, 2).trigger('click');

    const massRemoveButton = selectMassRemoveButton(wrapper);
    await massRemoveButton.triggerCustomEvent('click');

    expect(wrapper).toEmit(
      'remove-selected',
      [patterns[0], patterns[2]],
    );
  });

  it('Remove event emitted after trigger click on the remove button', async () => {
    const wrapper = snapshotFactory({
      propsData: {
        patterns,
        options: {
          page: 1,
          itemsPerPage: 10,
        },
        totalItems,
      },
    });

    await flushPromises();

    const removableRowIndex = 2;

    await selectRowRemoveButtonByIndex(wrapper, removableRowIndex).triggerCustomEvent('click');

    expect(wrapper).toEmit('remove', patterns[removableRowIndex]._id);
  });

  it('Update event emitted after trigger click on the remove button', async () => {
    const wrapper = snapshotFactory({
      propsData: {
        patterns,
        options: {
          page: 1,
          itemsPerPage: 10,
        },
        totalItems,
      },
    });

    await flushPromises();

    const editableRowIndex = 5;

    await selectRowEditButtonByIndex(wrapper, editableRowIndex).triggerCustomEvent('click');

    expect(wrapper).toEmit('edit', patterns[editableRowIndex]);
  });

  it('Renders `patterns-list` with default props', () => {
    const wrapper = snapshotFactory({
      propsData: {
        patterns,
        options: {},
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('Renders `patterns-list` with custom props', () => {
    const wrapper = snapshotFactory({
      propsData: {
        patterns,
        options: {
          page: 2,
          itemsPerPage: 10,
          search: 'Filter',
          sortBy: ['updated'],
          sortDesc: [true],
        },
        totalItems,
        pending: true,
        removable: true,
        updatable: true,
        duplicable: true,
        corporate: true,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
