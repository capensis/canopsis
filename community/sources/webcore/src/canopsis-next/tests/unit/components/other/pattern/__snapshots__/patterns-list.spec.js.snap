// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`patterns-list Renders \`patterns-list\` with custom props 1`] = `
<div
  class="c-advanced-data-table white"
  duplicable="true"
  removable="true"
  updatable="true"
>
  <div
    class="layout row wrap"
  >
    <div
      class="flex xs4"
    >
      <c-search-field-stub
        value="Filter"
      />
    </div>
    <div
      class="flex xs4"
      style="display: none;"
    >
      <c-action-btn-stub
        type="delete"
      />
    </div>
    <!---->
  </div>
  <div>
    <div
      class="v-table__overflow"
    >
      <table
        class="v-datatable v-table v-datatable--select-all theme--light"
      >
        <thead>
          <tr>
            <th>
              <div
                class="v-input v-input--selection-controls v-input--checkbox v-input--hide-details theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-input--selection-controls__input"
                    >
                      <input
                        aria-checked="false"
                        role="checkbox"
                        type="checkbox"
                        value=""
                      />
                      <div
                        class="v-input--selection-controls__ripple"
                      />
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        check_box_outline_blank
                      </i>
                    </div>
                  </div>
                </div>
              </div>
            </th>
            <th
              aria-label="Title: Not sorted. Activate to sort ascending."
              aria-sort="none"
              class="column sortable text-xs-left"
              role="columnheader"
              scope="col"
              tabindex="0"
            >
              Title
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
                style="font-size: 16px;"
              >
                arrow_upward
              </i>
            </th>
            <th
              aria-label="Type: Not sorted."
              aria-sort="none"
              class="column text-xs-left"
              role="columnheader"
              scope="col"
            >
              Type
            </th>
            <th
              aria-label="Last modified on: Sorted descending. Activate to remove sorting."
              aria-sort="descending"
              class="column active desc sortable text-xs-left"
              role="columnheader"
              scope="col"
              tabindex="0"
            >
              Last modified on
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
                style="font-size: 16px;"
              >
                arrow_upward
              </i>
            </th>
            <th
              aria-label="Last modified by: Not sorted. Activate to sort ascending."
              aria-sort="none"
              class="column sortable text-xs-left"
              role="columnheader"
              scope="col"
              tabindex="0"
            >
              Last modified by
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
                style="font-size: 16px;"
              >
                arrow_upward
              </i>
            </th>
            <th
              aria-label="Actions: Not sorted."
              aria-sort="none"
              class="column text-xs-left"
              role="columnheader"
              scope="col"
            >
              Actions
            </th>
          </tr>
          <tr
            class="v-datatable__progress"
          >
            <th
              class="column"
              colspan="6"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                class="v-progress-linear"
                role="progressbar"
                style="height: 2px;"
              >
                <div
                  class="v-progress-linear__background primary"
                  style="height: 2px; opacity: 0.3; width: 100%;"
                />
                <div
                  class="v-progress-linear__bar"
                >
                  <transition-stub
                    name="fade-transition"
                  >
                    <div
                      class="v-progress-linear__bar__indeterminate v-progress-linear__bar__indeterminate--active"
                    >
                      <div
                        class="v-progress-linear__bar__indeterminate long primary"
                      />
                      <div
                        class="v-progress-linear__bar__indeterminate short primary"
                      />
                    </div>
                  </transition-stub>
                  <transition-stub
                    name="slide-x-transition"
                  />
                </div>
              </div>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Alarm pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:48
              </span>
            </td>
            <td>
              author-0
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Entity pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:49
              </span>
            </td>
            <td>
              author-1
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Pbehavior pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:50
              </span>
            </td>
            <td>
              author-2
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Alarm pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:51
              </span>
            </td>
            <td>
              author-3
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Entity pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:52
              </span>
            </td>
            <td>
              author-4
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Pbehavior pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:53
              </span>
            </td>
            <td>
              author-5
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Alarm pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:54
              </span>
            </td>
            <td>
              author-6
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Entity pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:55
              </span>
            </td>
            <td>
              author-7
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Pbehavior pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:56
              </span>
            </td>
            <td>
              author-8
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Alarm pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:57
              </span>
            </td>
            <td>
              author-9
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Entity pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:58
              </span>
            </td>
            <td>
              author-10
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="6"
              tag="td"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <c-table-pagination-stub
    page="2"
    rows-per-page="10"
    total-items="11"
  />
</div>
`;

exports[`patterns-list Renders \`patterns-list\` with default props 1`] = `
<div
  class="c-advanced-data-table white"
>
  <div
    class="layout row wrap"
  >
    <div
      class="flex xs4"
    >
      <c-search-field-stub />
    </div>
    <div
      class="flex xs4"
      style="display: none;"
    >
      <c-action-btn-stub
        type="delete"
      />
    </div>
    <!---->
  </div>
  <div>
    <div
      class="v-table__overflow"
    >
      <table
        class="v-datatable v-table v-datatable--select-all theme--light"
      >
        <thead>
          <tr>
            <th>
              <div
                class="v-input v-input--selection-controls v-input--checkbox v-input--hide-details theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-input--selection-controls__input"
                    >
                      <input
                        aria-checked="false"
                        role="checkbox"
                        type="checkbox"
                        value=""
                      />
                      <div
                        class="v-input--selection-controls__ripple"
                      />
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        check_box_outline_blank
                      </i>
                    </div>
                  </div>
                </div>
              </div>
            </th>
            <th
              aria-label="Title: Sorted ascending. Activate to sort descending."
              aria-sort="ascending"
              class="column active asc sortable text-xs-left"
              role="columnheader"
              scope="col"
              tabindex="0"
            >
              Title
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
                style="font-size: 16px;"
              >
                arrow_upward
              </i>
            </th>
            <th
              aria-label="Type: Not sorted."
              aria-sort="none"
              class="column text-xs-left"
              role="columnheader"
              scope="col"
            >
              Type
            </th>
            <th
              aria-label="Last modified on: Not sorted. Activate to sort ascending."
              aria-sort="none"
              class="column sortable text-xs-left"
              role="columnheader"
              scope="col"
              tabindex="0"
            >
              Last modified on
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
                style="font-size: 16px;"
              >
                arrow_upward
              </i>
            </th>
            <th
              aria-label="Actions: Not sorted."
              aria-sort="none"
              class="column text-xs-left"
              role="columnheader"
              scope="col"
            >
              Actions
            </th>
          </tr>
          <tr
            class="v-datatable__progress"
          >
            <th
              class="column"
              colspan="5"
            />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Alarm pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:48
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Entity pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:49
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Pbehavior pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:50
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Alarm pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:51
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Entity pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:52
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Pbehavior pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:53
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Alarm pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:54
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Entity pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:55
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Pbehavior pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:56
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Alarm pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:57
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
          <tr>
            <td>
              <div
                class="layout c-checkbox-wrapper row justify-start"
              >
                <v-checkbox-functional-stub
                  hidedetails="true"
                  label=""
                  primary="primary"
                />
                <!---->
              </div>
            </td>
            <td>
              
            </td>
            <td>
              <span>
                Entity pattern
              </span>
            </td>
            <td>
              <span>
                04/03/2021 13:44:58
              </span>
            </td>
            <td>
              <div
                class="layout row"
              >
                <c-action-btn-stub
                  type="edit"
                />
                <c-action-btn-stub
                  type="delete"
                />
              </div>
            </td>
          </tr>
          <tr
            class="v-datatable__expand-row"
          >
            <transition-group-stub
              class="v-datatable__expand-col"
              colspan="5"
              tag="td"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <c-table-pagination-stub />
</div>
`;
